### PR TITLE
Issue #893: satelite projects should use latest checkstyle as soon at is appears in maven repo

### LINKED ISSUE #893: satelite projects should use latest checkstyle as soon at is appears in maven repo
Right now we do bump a version in a few projects https://github.com/checkstyle/checkstyle/wiki/How-to-make-a-release#update-for-released-version-reference right after release.

Let's find a way to define in pom.xml the latest snapshot and most recent version. And the latest released version when we define dependencies. To let us skip such updates at all.

CI of such projects should be able to get the latest version from https://repo1.maven.org/maven2/com/puppycrawl/tools/checkstyle/maven-metadata.xml and run projects with such version.

We can have in all such repos, a GitHub action that runs once a day check https://repo1.maven.org/maven2/com/puppycrawl/tools/checkstyle/maven-metadata.xml, put such values in required files, and send PRs with changes (PR is necessary as we need notice is any problems happens if we introduce breaking changes)

TASK:

- [ ] Version Bump in `sevntu-checks/pom.xml`
- [ ] Version Bump in `sevntu-checkstyle-sonar-plugin/pom.xml`

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Partially Fixes #893

### CODE REVIEW

File: .ci/bump-sonar-plugin-version.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -e
+
+POM_FILE=sevntu-checkstyle-sonar-plugin/pom.xml
```
Comment (romani) [inline .ci/bump-sonar-plugin-version.sh:4]: apply changes from https://github.com/checkstyle/contribution/pull/635#pullrequestreview-1079599409

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: .ci/bump-sonar-plugin-version.sh (hunk 1 above)
Comment (Rahulkhinchi03) [reply .ci/bump-sonar-plugin-version.sh]: Changes Done

File: .ci/bump-cs-version-in-sonar-plugin.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -e
+
+if [[ -z $1 ]]; then
+  echo "ERROR: version is not set"
+  echo "Usage: bump-cs-version-in-gradle-project.sh <version>"
```
Comment (romani) [inline .ci/bump-cs-version-in-sonar-plugin.sh:6]: fix name

File: .ci/bump-cs-version-in-sonar-plugin.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  exit 1
+fi
+
+VERSION=$1
+SONAR_PLUGIN_FILE=sevntu-checkstyle-sonar-plugin
```
Comment (romani) [inline .ci/bump-cs-version-in-sonar-plugin.sh:11]: file is pom.xml

**Comment by Rahulkhinchi03** [comment]
TESTING:

```
rahul@rk7:~/Desktop/opensource/sevntu.checkstyle$ ./.ci/bump-cs-version-in-sonar-plugin.sh 10.5
[WARNING] 
[WARNING] Some problems were encountered while building the effective settings
[WARNING] Expected root element 'settings' but found 'servers' (position: START_TAG seen <servers>... @1:9)  @ /home/rahul/.m2/settings.xml, line 1, column 9
[WARNING] 
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] 
[INFO] ----< com.github.sevntu-checkstyle:sevntu-checkstyle-sonar-plugin >-----
[INFO] Building SevNTU Checkstyle Sonar Extension Plugin 1.42.0
[INFO] ----------------------------[ sonar-plugin ]----------------------------
[INFO] 
[INFO] --- versions-maven-plugin:2.11.0:set-property (default-cli) @ sevntu-checkstyle-sonar-plugin ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.874 s
[INFO] Finished at: 2022-08-27T22:08:16+05:30
[INFO] ------------------------------------------------------------------------
Version updated to 10.5 at sevntu-checkstyle-sonar-plugin
rahul@rk7:~/Desktop/opensource/sevntu.checkstyle$ 
```

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: .ci/bump-cs-version-in-sonar-plugin.sh (hunk 3 above)
Comment (Rahulkhinchi03) [reply .ci/bump-cs-version-in-sonar-plugin.sh:11]: Ya but the project has only one pom file so it works well without adding `pom.xml`

File: .ci/bump-cs-version-in-sonar-plugin.sh (hunk 2 above)
Comment (Rahulkhinchi03) [reply .ci/bump-cs-version-in-sonar-plugin.sh]: Changes Done

File: .ci/bump-cs-version-in-sonar-plugin.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+if [[ -z $1 ]]; then
+  echo "ERROR: version is not set"
+  echo "Usage: bump-cs-version-in-sonar-plugin.sh <version>"
+  exit 1
+fi
+
+VERSION=$1
+SONAR_PLUGIN_FILE=sevntu-checkstyle-sonar-plugin
+
+cd $SONAR_PLUGIN_FILE
```
Comment (romani) [inline .ci/bump-cs-version-in-sonar-plugin.sh:13]: `cd` to  file is nonsense

please remove this variable and use actual values you need.

File: .ci/bump-cs-version-in-sonar-plugin.sh (hunk 3 above)
Comment (romani) [reply .ci/bump-cs-version-in-sonar-plugin.sh:11]: see comment below

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: .ci/bump-cs-version-in-sonar-plugin.sh (hunk 4 above)
Comment (Rahulkhinchi03) [reply .ci/bump-cs-version-in-sonar-plugin.sh]: Changes Done

File: .ci/bump-cs-version-in-sonar-plugin.sh (hunk 3 above)
Comment (Rahulkhinchi03) [reply .ci/bump-cs-version-in-sonar-plugin.sh]: Changes Done

**Review by romani — APPROVED** [review]
ok to merge

**Pull request merged**